[invalid-name] Failure to recognize exclusive assignment for a value assigned a non-constant

### Bug description

```python
def a():
    return 1, 2, 3

if a():
    BBB = a()
else:
    BBB = 500
```

### Configuration

```ini

```

### Command used

```shell
pylint a.py
```

### Pylint output

```python
invalid-name
```

### Expected behavior

Thanks @nmlorg for the report

### Pylint version

```shell
main
```

### OS / Environment

_No response_

### Additional dependencies

```python

```

The [code](https://github.com/nmlorg/metabot/blob/bcdd3bd7671bfd543a82717a38a69f01a91ecca9/metabot/util/geoutil.py#L13-L18) I'm actually worried about is:
```py
try:
    _CLIENT_KEY = next(open('config/google_maps_apikey')).strip()
except IOError:
    _CLIENT = None
else:
    _CLIENT = googlemaps.Client(key=_CLIENT_KEY)
```
which is so similar to the reported code I added the example with `if`/`else` just to make it clear it's not specific to `try`/`except`. I'm guessing my issue is that it's `try`/`except`/`else`, so just want to make sure that's covered too!

<br>
<br>

```
$ git clone https://github.com/nmlorg/metabot pylint-10745
$ cd pylint-10745
$ git checkout bcdd3bd
$ python3 -m venv .venv
$ . .venv/bin/activate
$ pip install -U -e .[dev]
$ pylint metabot
************* Module util.geoutil
metabot/util/geoutil.py:16:4: C0103: Variable name "_CLIENT" doesn't conform to snake_case naming style (invalid-name)
metabot/util/geoutil.py:18:4: C0103: Variable name "_CLIENT" doesn't conform to snake_case naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 9.99/10 (previous run: 9.99/10, +0.00)

$ pip install -U pylint==3.3.9
$ pylint metabot

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 9.99/10, +0.01)
```

--- a/doc/whatsnew/4/4.0/index.rst
+++ b/doc/whatsnew/4/4.0/index.rst
@@ -74,6 +74,24 @@ to your liking.
 
 .. towncrier release notes start
 
+What's new in Pylint 4.0.4?
+--------------------------------
+Release date: 2025-11-30
+
+
+False Positives Fixed
+---------------------
+
+- Fixed false positive for ``invalid-name`` where module-level constants were incorrectly classified as variables when a class-level attribute with the same name exists.
+
+  Closes #10719 (`#10719 <https://github.com/pylint-dev/pylint/issues/10719>`_)
+
+- Fix a false positive for ``invalid-name`` on an UPPER_CASED name inside an ``if`` branch that assigns an object.
+
+  Closes #10745 (`#10745 <https://github.com/pylint-dev/pylint/issues/10745>`_)
+
+
+
 What's new in Pylint 4.0.3?
 ---------------------------
 Release date: 2025-11-13

--- a/doc/whatsnew/fragments/10719.false_positive
+++ b/doc/whatsnew/fragments/10719.false_positive
@@ -1,3 +1,0 @@
-Fixed false positive for ``invalid-name`` where module-level constants were incorrectly classified as variables when a class-level attribute with the same name exists.
-
-Closes #10719

--- a/doc/whatsnew/fragments/10745.false_positive
+++ b/doc/whatsnew/fragments/10745.false_positive
@@ -1,4 +1,0 @@
-Fix a false positive for ``invalid-name`` on an UPPER_CASED name inside an
-``if`` branch that assigns an object.
-
-Closes #10745

--- a/pylint/__pkginfo__.py
+++ b/pylint/__pkginfo__.py
@@ -9,7 +9,7 @@ It's updated via tbump, do not modify.
 
 from __future__ import annotations
 
-__version__ = "4.0.3"
+__version__ = "4.0.4"
 
 
 def get_numversion_from_version(v: str) -> tuple[int, int, int]:

--- a/tbump.toml
+++ b/tbump.toml
@@ -1,7 +1,7 @@
 github_url = "https://github.com/pylint-dev/pylint"
 
 [version]
-current = "4.0.3"
+current = "4.0.4"
 regex = '''
 ^(?P<major>0|[1-9]\d*)
 \.

